addons/payment_mollie/models/payment_transaction.py

```diff
@@ -62,7 +63,7 @@ def _mollie_prepare_payment_request_payload(self):
             self.currency_id.name, self.currency_id.decimal_places
         )
 
-        return {
+        payload = {
```

Comment:
No need to save it in payload anymore
```suggestion
       return {
```